cmd/monitor/main.go

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/sigstore/rekor-monitor/pkg/rekor"
+	"github.com/sigstore/rekor-monitor/pkg/util/file"
+	"github.com/sigstore/rekor/pkg/client"
+	"github.com/sigstore/rekor/pkg/util"
+	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/release-utils/version"
+)
+
+// Default values for monitoring job parameters
+const (
+	publicRekorServerURL     = "https://rekor.sigstore.dev"
+	logInfoFileName          = "logInfo.txt"
+	outputIdentitiesFileName = "identities.txt"
+)
+
+// This main function performs a periodic identity search.
+// Upon starting, any existing latest snapshot data is loaded and the function runs
+// indefinitely to perform identity search for every time interval that was specified.
+func main() {
+	// Command-line flags that are parameters to the verifier job
+	configFilePath := flag.String("config-file", "", "path to yaml configuration file containing identity monitor settings")
+	once := flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
+	flag.Parse()
+
+	if configFilePath == nil {
+		log.Fatalf("empty configuration file path")
+	}
+
+	readConfig, err := os.ReadFile(*configFilePath)
+	if err != nil {
+		log.Fatalf("error reading from identity monitor configuration file: %v", err)
+	}
+
+	configString := string(readConfig)
+	var config IdentityMonitorConfiguration
+	if err := yaml.Unmarshal([]byte(configString), &config); err != nil {
+		log.Fatalf("error parsing identities: %v", err)
+	}
+
+	rekorClient, err := client.GetRekorClient(config.ServerURL, client.WithUserAgent(strings.TrimSpace(fmt.Sprintf("rekor-monitor/%s (%s; %s)", version.GetVersionInfo().GitVersion, runtime.GOOS, runtime.GOARCH))))
+	if err != nil {
+		log.Fatalf("getting Rekor client: %v", err)
+	}
+
+	if config.StartIndex == nil || config.EndIndex == nil {
```

Comment:
This loop is missing writing the end index or the latest checkpoint to a file, so for a future invocation of the binary, we'd start from the same index, right?

Given that we don't actually need checkpoints, I wonder if we can simplify this if we instead propose a very basic structure to hold the last successfully verified index like `{last_index: x}` (and in the future, it could also include all indices that could not be verified, the ones we will skip over).

Then it becomes something like:

* Run all of the below in a loop:
  * If you don't specify a start index, read from file
  * If you don't specify an end index, ask for the latest checkpoint
  * If end < start, fail early
  * Run `IdentitySearch`
  * Persist end index in file
  * If `once` or you specified an end index, exit (WDYT about the latter part? I think if you say you want to check only up to X, then that is implicitly asking to be run once)

Reply:
Agree with all of this- I think `once` can default to true, and in the case that a user does want to initially check up to the end index but continue to leave the monitor running, that option can be left open to them